test/pickfirst_test.go

```diff
@@ -377,16 +379,17 @@ func (s) TestPickFirst_ShuffleAddressList(t *testing.T) {
 	const serviceConfig = `{"loadBalancingConfig": [{"pick_first":{ "shuffleAddressList": true }}]}`
 
 	// Install a shuffler that always reverses two entries.
-	origShuf := internal.ShuffleAddressListForTesting
-	defer func() { internal.ShuffleAddressListForTesting = origShuf }()
-	internal.ShuffleAddressListForTesting = func(n int, f func(int, int)) {
+	newShuffler := func(n int, f func(int, int)) {
 		if n != 2 {
 			t.Errorf("Shuffle called with n=%v; want 2", n)
 			return
 		}
 		f(0, 1) // reverse the two addresses
 	}
-
+	origShuf := internal.ShuffleAddressListForTesting.(func(sf shuffler) shuffler)(newShuffler)
+	defer func() {
+		internal.ShuffleAddressListForTesting.(func(sf shuffler) shuffler)(origShuf)
+	}()
```

Comment:
I like the idea of returning the shuffle func to the original value at the end of the test. That was something I did not think about when I made the comment on the original PR. But I don't like the idea of using the same function `internal.ShuffleAddressListForTesting` to do it because it looks a little ugly/weird to reset the thing using a `ForTesting` function.

What do you think about having two internal functions instead:
- `ShuffleAddressListForTesting` to override the shuffle func
- `RevertAddressListShuffleFunc` to revert to the default

I feel the above approach will make the test more readable.

Reply:
Changed.